### DISCUSSION TITLE
topaz denoise, what are appropriate input formats?

### DISCUSSION THREAD

**Discussion by larsonmattr** [tbepler/topaz#101] [Q&A] [answered]
Hi,

 I am trying to use `topaz denoise` with a movie taken in SerialEM (.st).  I am using v 0.2.5.  

 With original .st files, topaz denoise --patch-size 1024 -o Test_denoised/ Test/*.st,
 -> There is an error PIL.UnidentifiedImageError: cannot identify image file

 When I convert first to .mrc via the EMAN2 `e2proc3d.py`program, and use then: 
  topaz denoise --patch-size 1024 -o Test_denoised/ Test/*.mrc, 
 -> There is an error TypeError: Cannot handle this data type: (1, 1, 5760), <f4

 The movies are recorded on a Gatan K3 detector and the output from EMAN `header` is:

 RO image file on unit   1 : Test/009.mrc     Size=    4695571 K

 Number of columns, rows, sections .....    5760    4092      51
 Map mode ..............................    2   (32-bit real)              
 Start cols, rows, sects, grid x,y,z ...    0     0     0    5760   4092     51
 Pixel spacing (Angstroms)..............   10.25      10.25      10.25    
 Cell angles ...........................   90.000   90.000   90.000
 Fast, medium, slow axes ...............    X    Y    Z
 Origin on x,y,z .......................    0.000       0.000       0.000    
 Minimum density .......................   0.0000    
 Maximum density .......................   32767.    
 Mean density ..........................   2024.3    
 RMS deviation from mean................   832.76    
 tilt angles (original,current) ........   0.0   0.0   0.0   0.0   0.0   0.0
 Space group,# extra bytes,idtype,lens .        0        0        0        0

---

 It sounded like the correct input for `topaz denoise` would be MRC movie formats with > 1 subframe, so the algorithm can use both even and odd frames as separate groups in denoising. Is this correct?  Could the issue be the `Map mode` of the movie?

 Thanks,
   Matt Larson

**Comment by Guillawme** [comment, marked as answer]
As I understand it, the input must be movies when you want to _train_ a denoiser. But if you want to denoise micrographs using an _already trained_ denoiser, then the input must be micrographs (so, the output of motion correction, with a single frame). I only have limited experience using `topaz denoise`, but it always gave good results, so I never needed to train a denoiser on my own movies.

Good luck!

**Reply by tbepler** [reply]
Exactly, `topaz denoise` expects micrographs as inputs. Even for training, the micrographs still need to be processed from the frames.
